### ISSUE TITLE
Oculus Rift Template

### ISSUE THREAD

**Issue by tomspilman** [mono/MonoGame#4409]
We should consider adding template with out of the box Oculus Rift support.

Doesn't need to be very complex... just a spinning cube...  but should show we support it.

Likely this template should be implemented against the new project creation tool in https://github.com/mono/MonoGame/pull/4295.

**Comment by iamthefaker** [comment]
How is VR/MR support progressing in monogame? I realise that VR/MR is only just getting started for end-users, but for developers to arrive with software at the same time that users arrive with hardware, need to have support for VR/MR before that. Otherwise we will be playing catch up. Unity already appears to have support for Occulus, GearVR and Hololens. And the buzz lately has been growing ... https://www.youtube.com/watch?v=xvyMoKI9zfg

**Comment by DoogeJ** [comment]
@iamthefaker check out http://community.monogame.net/t/so-vr-ar/6839/3

Quoting @tomspilman :

> There is nothing special needed from MonoGame to render to Oculus or other VR goggles. You just render the scene twice and apply a shader to distort the image. This was covered fairly well a few years ago:
> https://visualstudiomagazine.com/articles/2013/10/30/virtual-reality--in-dotnet-part-1.aspx

**Comment by DDReaper** [comment]
TBH, it would be better to have an abstraction (or extension project) for VR solutions, rather than tying to any one platform.

**Comment by YugeshN** [comment]
I agree with @DDReaper ... this should be abstracted to support multiple platforms / systems / hardware.
